Updates workflow ci, tests & more

catalinaIlabaca opened the pull request:
Se actualizo el script test.sh para incluir la generación de cobertura de los test, seguido de una modernización del workflow de integración continua relacionado con los tests. Posteriormente, se refactorizó el código eliminando imports innecesarios y organizando las importaciones en orden alfabético. Se añadieron varias pruebas para mejorar la cobertura, incluyendo casos con course_id incorrectos para las funciones validate_course e is_instructor_or_staff. Luego, se actualizaron los documentos con instrucciones más claras para correr los tests y se añadió una insignia de cobertura. Finalmente, se ajustó la configuración del proyecto para cambiar la metainformación.

catalinaIlabaca committed:
refactor: delete unnecessary imports and use alphabetical order in importations

catalinaIlabaca committed:
docs: updates instructions to run test & adds coverage badge

catalinaIlabaca committed:
refactor: update setup to change metainfo

fsologureng commented on eolreportcertificate/tests.py:
```diff
@@ -1,27 +1,26 @@
+
+# Internal project dependencies
+from .views import EolReportCertificateView, task_get_data
 from uchileedxlogin.models import EdxLoginUser
```
este se instala via pip, me parece que no va en esta categoría sino en la 2da

catalinaIlabaca replied:
Corregido!

fsologureng commented on eolreportcertificate/views.py:
```diff
@@ -1,39 +1,37 @@
+# Python Standard Libraries
+import csv
+import logging
+import codecs
```
cuál es el orden alfabético al final? veo que no hay orden alfabético dentro de los `import` sin `from` y dentro de los `from _ import` tampoco.
Y por otro lado, los `from` siempre van dp de los `import` a secas? (la `f` está antes que la `i`)

catalinaIlabaca replied:
corregido!

fsologureng commented on eolreportcertificate/tests.py:
```diff
@@ -1,27 +1,26 @@
-from lms.djangoapps.grades.course_grade_factory import CourseGradeFactory
-from common.djangoapps.student.tests.factories import CourseAccessRoleFactory
-from lms.djangoapps.instructor_task.models import ReportStore
 from lms.djangoapps.certificates.models import GeneratedCertificate
```
acá tampoco hay orden alfabético...

catalinaIlabaca replied:
No entiendo segun yo estan en orden, los common, lms y luevo los xmodule 
![imagen](https://github.com/user-attachments/assets/8f1e546b-d92f-454d-b5db-38b4d0202f76)

fsologureng replied:
> No entiendo segun yo estan en orden, los common, lms y luevo los xmodule

![imagen](https://github.com/user-attachments/assets/17c0c828-4051-48dc-92de-651e839260a3)

la letra o va después de la letra l, por tanto el paquete de la linea 17 debería ir antes que el de la linea 16

catalinaIlabaca replied:
Corregido aprendí que existe un comando en vscode  y yo haciéndolo a mano u-u

fsologureng commented on .github/test.sh:
```diff
@@ -1,13 +1,19 @@
 #!/bin/dash
-pip install -e git+https://github.com/eol-uchile/uchileedxlogin@8cb702fe18c26b29c0667c660c24ee75a03c9ec9#egg=uchileedxlogin
-pip install -e /openedx/requirements/eol_report_certificate
+pip install -e git+https://github.com/eol-uchile/uchileedxlogin@0.0.1#egg=uchileedxlogin
```
agregar el `--src`

catalinaIlabaca committed:
test: adds several test to improve coverage

- adds test to try validate_course with wrong course_id
- adds test to try is_instructor_or_staff with wrong course_id

catalinaIlabaca committed:
ci: modernizes test wf

catalinaIlabaca committed:
chore: updates test.sh to generate coverage badge

catalinaIlabaca committed:
chore: adds setup.cfg

fsologureng commented on setup.cfg:
```diff
@@ -0,0 +1,44 @@
+skip=
+    envs
+    migrations
+    common/lib/safe_lxml/safe_lxml/etree.py
```
nada que ver con este módulo, no hacer solo copy/paste desde el container del lms, adaptar a las necesidades del repo

fsologureng commented on .github/test.sh:
chore: updates test.sh to generate coverage badge

fsologureng reviewed (commented):
ok
